lib/ddtrace/contrib/faraday/middleware.rb

```diff
@@ -0,0 +1,52 @@
+require 'faraday'
+require 'ddtrace/ext/http'
+require 'ddtrace/ext/net'
+
+module Datadog
+  module Contrib
+    module Faraday
+      # Middleware implements a faraday-middleware for ddtrace instrumentation
+      class Middleware < ::Faraday::Middleware
+        def initialize(*)
+          @tracer = try_pin(:tracer, Datadog.tracer)
```

Comment:
theoretically, the Faraday patch method must be called before initializing the middleware (it should be a right assumption), so the `Pin.get_from(::Faraday)` should always return a valid `Pin` instance (or `nil` if not patched). If that's true, can't we rely directly in the Faraday Pin instead of expecting these things set at initialization time?

My main concern is what happens if users (after initializing Faraday), use the `Pin` to change the service name. Accessing the `Pin` object and checking if it's available + enabled, should avoid any order issues.